### PR TITLE
Updation and Addition in the Sybil MVP contract, interface and test file

### PR DESCRIPTION
**Pull request by DarkKnight3074**
This PR introduces updates and additions to the Sybil MVP contract, its interface and its test cases.

Changes in Sybil.sol:

- Updated the deposit, exit, and explode functions by adding necessary checks.
- Renamed and updated the createAccount function to createAccountDeposit.
- Added two new functions: vouch and unvouch.

Changes in the Interface:

- Added the vouch and unvouch functions.
- Renamed createAccount to createAccountDeposit.
- Updated the custom errors.

Test Scenario covered:
- Happy Path
- Invalid Scenarios
- Edge Cases

### CODE REVIEW

File: contracts/src/interfaces/IMVPSybil.sol [hunk 1]
```diff
@@ -3,29 +3,22 @@ pragma solidity 0.8.23;
...
+    error InvalidFromIdx();
+    error InvalidToIdx();
 
     // Initialization function
     function initialize(
         address[] memory verifiers,
         uint256[] memory maxTxs,
         uint256[] memory nLevels,
-        uint8 _forgeL1L2BatchTimeout, 
+        uint8 _forgeL1L2BatchTimeout,
```
Comment (amantokamak) [inline contracts/src/interfaces/IMVPSybil.sol]: It should be L1 only or remove both as we are only dealing with L1

**Review by amantokamak — CHANGES_REQUESTED** [review]
Please go through the checks that you added in functions. I saw all the changes and compared it with Jamie document, I found there are incorrect checks that you added.

File: contracts/src/interfaces/IMVPSybil.sol (hunk 1 above)
Comment (DarkKnight3074) [reply contracts/src/interfaces/IMVPSybil.sol]: Okay, I will do that.

**Comment by DarkKnight3074** [comment]
> Please go through the checks that you added in functions. I saw all the changes and compared it with Jamie document, I found there are incorrect checks that you added.

I will make the necessary changes.

**Review by amantokamak — APPROVED** [review]
Seems fine now
